tests/client/amo/components/TestOverallRating.js

```diff
@@ -182,31 +253,254 @@ describe('OverallRating', () => {
+              ...fakeReview,
+              userId,
+              version: { ...fakeAddon.current_version, id: versionId },
+            },
+          ],
+        };
+        mockApi.expects('getUserReviews').returns(Promise.resolve(response));
+
+        return loadSavedRating({ userId, versionId })
+          .then(() => {
+            assert.equal(dispatch.called, true);
+            assert.deepEqual(dispatch.firstCall.args[0], setReview({
+              addonId: fakeReview.addon.id,
+              rating: fakeReview.rating,
+              versionId,
+              userId,
+            }));
+          });
       });
 
-      it('sets the userId property from the state', () => {
-        const authState = {
-          token: signedInApiState.token,
-          userId: 91234,
+      it('expects to only find one matching review', () => {
+        const response = { results: [fakeReview, fakeReview] };
+        mockApi.expects('getUserReviews').returns(Promise.resolve(response));
+
+        return loadSavedRating()
+          .then(() => {
+            throw new Error('unexpected success');
+          })
+          .catch((error) => {
+            assert.match(error.message, /received more than one review/);
+          });
+      });
+
+      it('does nothing when there are not any matching reviews', () => {
+        const addonId = 8765;
+        const response = {
+          results: [
+            // All of these have the wrong IDs:
+            { ...fakeReview, addon: { ...fakeAddon, id: 123 } },
+            { ...fakeReview, addon: { ...fakeAddon, id: 321 } },
+          ],
         };
-        const props = mapStateToProps({ auth: authState });
-        assert.equal(props.userId, 91234);
+        mockApi.expects('getUserReviews').returns(Promise.resolve(response));
+
+        return loadSavedRating({ addonId })
+          .then(() => {
+            assert.equal(dispatch.called, false);
+          });
+      });
+
+      it('does nothing when the API returns no reviews at all', () => {
+        const response = { results: [] };
+        mockApi.expects('getUserReviews').returns(Promise.resolve(response));
+
+        return loadSavedRating()
+          .then(() => {
+            assert.equal(dispatch.called, false);
+          });
       });
     });
   });
+
+  describe('mapStateToProps', () => {
+    let store;
+
+    beforeEach(() => {
+      store = createStore();
+    });
+
+    function getMappedProps({
+      state = store.getState(),
+      componentProps = {
+        addonId: fakeAddon.id,
+        version: fakeAddon.current_version,
+      },
+    } = {}) {
+      return mapStateToProps(state, componentProps);
+    }
+
+    function signIn({ userId = 98765 } = {}) {
+      store.dispatch(setJWT(userAuthToken({
+        user_id: userId,
+      })));
+    }
+
+    it('copies api state to props', () => {
+      signIn();
+
+      const state = store.getState();
+      assert(state.api.token, 'a valid token exists in state');
```

Comment:
I don't think this assertion needs to happen for every test. I only added it here because the assertion below it has the possibly to silently fail if `state.api` is falsey. If we introduce static Flow typing then that wouldn't be an issue!